x-pack/plugins/features/server/plugin.ts

```diff
@@ -103,13 +107,8 @@ export class FeaturesPlugin
       featureRegistry: this.featureRegistry,
     });
 
-    const getFeaturesUICapabilities = () =>
-      uiCapabilitiesForFeatures(
-        this.featureRegistry.getAllKibanaFeatures(),
-        this.featureRegistry.getAllElasticsearchFeatures()
-      );
-
-    core.capabilities.registerProvider(getFeaturesUICapabilities);
+    // capabilities provider wil never be called before plugin start
+    core.capabilities.registerProvider(() => this.capabilities!);
```

Comment:
Before someone asks: we are deepCloning capabilities when toggling them, so returning the same object for every calls is fine 😄

Reply:
I assume we don't have any external consumers of `getFeaturesUiCapabilitites` from the setup contract. Even if there are, they should have noticed Kibana crashing if calling it before setup finished.